tests/api_connexion/endpoints/test_extra_link_endpoint.py

```diff
@@ -15,24 +15,198 @@
+from urllib.parse import quote
 
-import pytest
+from parameterized import parameterized
+from test_utils.mock_plugins import mock_plugin_manager
 
+from airflow import DAG
+from airflow.models.baseoperator import BaseOperatorLink
+from airflow.models.dagrun import DagRun
+from airflow.models.xcom import XCom
+from airflow.plugins_manager import AirflowPlugin
+from airflow.providers.google.cloud.operators.bigquery import BigQueryExecuteQueryOperator
+from airflow.utils.dates import days_ago
+from airflow.utils.session import provide_session
+from airflow.utils.timezone import datetime
+from airflow.utils.types import DagRunType
 from airflow.www import app
+from tests.test_utils.db import clear_db_runs, clear_db_xcom
 
 
 class TestGetExtraLinks(unittest.TestCase):
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
-        cls.app = app.create_app(testing=True)  # type:ignore
+        with mock.patch.dict('os.environ', SKIP_DAGS_PARSING='True'):
+            cls.app = app.create_app(testing=True)  # type:ignore
+
+    @provide_session
+    def setUp(self, session) -> None:
+        self.now = datetime(2020, 1, 1)
+
+        clear_db_runs()
+        clear_db_xcom()
+
+        self.dag = self._create_dag()
+        self.app.dag_bag.dags = {self.dag.dag_id: self.dag}  # type: ignore  # pylint: disable=no-member
+        self.app.dag_bag.sync_to_db()   # type: ignore  # pylint: disable=no-member
+
+        dr = DagRun(
+            dag_id=self.dag.dag_id,
+            run_id="TEST_DAG_RUN_ID",
+            execution_date=self.now,
+            run_type=DagRunType.MANUAL.value,
+        )
+        session.add(dr)
+        session.commit()
 
-    def setUp(self) -> None:
         self.client = self.app.test_client()  # type:ignore
 
-    @pytest.mark.skip(reason="Not implemented yet")
+    def tearDown(self) -> None:
+        super().tearDown()
+        clear_db_runs()
+        clear_db_xcom()
+
+    @staticmethod
+    def _create_dag():
+        with DAG(dag_id="TEST_DAG_ID", default_args=dict(start_date=days_ago(2),)) as dag:
+            BigQueryExecuteQueryOperator(task_id="TEST_SINGLE_QUERY", sql="SELECT 1")
+            BigQueryExecuteQueryOperator(task_id="TEST_MULTIPLE_QUERY", sql=["SELECT 1", "SELECT 2"])
+        return dag
+
+    @parameterized.expand(
+        [
+            (
+                "/api/v1/dags/INVALID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_SINGLE_QUERY/links",
+                'DAG not found'
+            ),
+            (
+                "/api/v1/dags/TEST_DAG_ID/dagRuns/INVALID/taskInstances/TEST_SINGLE_QUERY/links",
+                "DAG Run not found"
+            ),
+            (
+                "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/INVALID/links",
+                "Task not found"
+            ),
+        ]
+    )
+    def test_should_response_404_on_invalid_task_id(self, url, expected_title):
+        response = self.client.get(url)
```

Comment:
```suggestion
    @parameterized.expand(
        [
            (
                "invalid dag_id",
                "/api/v1/dags/INVALID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_SINGLE_QUERY/links",
                'DAG not found'
            ),
            (
                "invalid run_id",
                "/api/v1/dags/TEST_DAG_ID/dagRuns/INVALID/taskInstances/TEST_SINGLE_QUERY/links",
                "DAG Run not found"
            ),
            (
                "invalid task_id",
                "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/INVALID/links",
                "Task not found"
            ),
        ]
    )
    def test_should_response_404_on(self, name, url, expected_title):
        del name
        response = self.client.get(url)
```
These tests are dealing with multiple invalid id(s), this is IMO more descriptive, what do you think ?

Reply:
In many places, we don't use names. We already have one name that allows you to identify a specific test, then the number allows you to specify a specific variant.  Many names will limit the number of tests. What I love in parameterized is the ability to generate many cases in a simple way. It is possible that some cases may overlap, but this may change in the future.